File: Placement Sheet/String/Service Bases/Encode_The_Message.js
function encode(message) {
    if (message == null || message.length == 0) {
        return null;
    }

    let str = "";
    let strArr = message.split("");
    let i = 0;

    while (i < strArr.length) {
        let currElm = strArr[i];
        let count = 1;

        while (i + 1 < strArr.length && currElm == strArr[i + 1]) {
            count++;
            i++;
        }

        str += currElm + count;
        i++;
    }

    return str;
}

let message = "aabcccdeee";
console.log(`Encoded message: ${encode(message)}`);

File: Placement Sheet/String/Service Bases/Encode_The_Message.java
import java.util.Scanner;

public class Encode_The_Message {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the string: ");
        String str = input.next();
        input.close();
        System.out.println("Real message: " + str);
        System.out.println("Encoded message: " + encode(str));
    }

    private static String encode(String message) {
        if (message == null || message.isBlank()) {
            return null;
        }

        StringBuilder builder = new StringBuilder();
        char[] strArr = message.toCharArray();

        int i = 0;

        while (i < strArr.length) {
            char currElm = strArr[i];
            int count = 1;
            while (i + 1 < strArr.length && strArr[i + 1] == currElm) {
                count++;
                i++;
            }
            builder.append(currElm + String.valueOf(count));
            i++;
        }

        return builder.toString();
    }
}
